### PR TITLE
Observation model

### PR DESCRIPTION
**Pull request by t-kimber**
## Description
This PR deals with the observation model implemented in kinoml (theory and implementation)

## Todos
 - [ ] check observation model in the api for all 4 measurement types (and measurement units)
    - [x] pIC50
    - [x] pKi
    - [x] pKd
    - [ ] %displ.
- [x] adapt kinoml workflow (with theory and image) 

## Question
- Where should the explanation (the theory behind the observation model) live? 

## Status
- [ ] Ready to go

### CODE REVIEW

**Comment by t-kimber** [comment]
**TO BE CHECKED!**

With the new definition of the observation model, see https://github.com/openkinome/kinoml/blob/observation-model/kinoml/core/measurements.py defined by the function `_observation_model_pytorch`
which return
`-(dG_over_KT + numpy.log((1 + substrate_conc / michaelis_constant) * inhibitor_conc)) * 1 / numpy.log(10)`,
the workflow notebook (https://github.com/openkinome/kinoml/blob/observation-model/kinoml/workflows/KinoML_workflow.ipynb) runs perfectly. However, the API example notebook (https://github.com/openkinome/kinoml/blob/observation-model/examples/ChEMBL.ipynb) gives the error 
`RuntimeError: Found dtype Double but expected Float`when trying to execute the optimization loop.

File: kinoml/core/measurements.py [hunk 1]
```diff
@@ -118,17 +118,17 @@ def check(self):
         assert (0 <= self.values <= 100).all(), "One or more values are not in [0, 100]"
 
     @staticmethod
-    def _observation_model_pytorch(dG_over_KT, inhibitor_conc=1e-6, **kwargs):
+    def _observation_model_pytorch(dG_over_KT, inhibitor_conc=1, **kwargs):
         # FIXME: this might be upside down -- check!
         import torch
 
-        return 100 * (1 - 1 / (1 + inhibitor_conc / torch.exp(dG_over_KT)))
+        return 100 * (1 - 1 / (1 +1  / torch.exp(dG_over_KT)))
```
Comment (jaimergp) [inline kinoml/core/measurements.py:125]: Why do we need to hardcode this? Don't we have the argument in the function? Also, why are we changing units now?

Comment (t-kimber) [reply kinoml/core/measurements.py:125]: I need to speak with John for this observation model. I thought that this measurement type was in the clear, but I realized at the end of last week that something is fishy and I need to loop in John asap.

File: kinoml/core/measurements.py [hunk 2]
```diff
@@ -200,22 +204,19 @@ class pKdMeasurement(BaseMeasurement):
...
     """
 
     @staticmethod
-    def _observation_model_pytorch(dG_over_KT, inhibitor_conc=1e-6, **kwargs):
+    def _observation_model_pytorch(dG_over_KT, inhibitor_conc=1, **kwargs):
         import torch
+        import numpy
 
-        return -torch.log10(torch.exp(dG_over_KT) * inhibitor_conc * 1e-9)
+        return -(dG_over_KT + numpy.log(inhibitor_conc)) * 1 / numpy.log(10)
```
[the remark this answers is not in the document]
Comment (t-kimber) [reply kinoml/core/measurements.py:219]: See reference for 1 [M] concentration: 
https://onlinelibrary.wiley.com/doi/epdf/10.1002/wcms.1455

**Comment by jaimergp** [comment]
> RuntimeError: Found dtype Double but expected Float

This should be fixed by last commits.

**Pull request merged**